drizzlepac/generate_final_product_filenames.py

```diff
@@ -1,15 +1,160 @@
-
 """This script is development space for code that will be most likely
     absorbed into a larger script later on.
 
 """
 
-def run_generator(product_category,obs_info):
+from astropy.table import Table, Column
+
+# Define information/formatted strings to be included in output dict
+SEP_STR = 'single exposure product {:02d}'
+FP_STR = 'filter product {:02d}'
+TDP_STR = 'total detection product {:02d}'
+
+# Define the mapping between the first character of the filename and the associated instrument
+INSTRUMENT_DICT = {'i': 'WFC3', 'j': 'ACS', 'o': 'STIS', 'u': 'WFPC2', 'x': 'FOC', 'w': 'WFPC'}
+
+def interpret_obset_input(results):
     """
-    This is the main calling subroutine. It decides which filename generation subroutine should be run based on the
-    input product_category, and then passes the information stored in input obs_info to the subroutine so that the
-    appropriate filenames can be generated.
+    Interpret the database query for a given obset to prepare the returned
+    values for use in generating the names of all the expected output products.
+
+    Input will have format of:
+        ia1s70jtq_flt.fits,11150,A1S,70,70,149.232269,F110W,IR,/ifs/archive/ops/hst/public/ia1s/ia1s70jtq/ia1s70jtq_flt.fits
+        ia1s70iwq_flt.fits,11150,A1S,70,70,0.91161000000000003,F160W,IR,/ifs/archive/ops/hst/public/ia1s/ia1s70iwq/ia1s70iwq_flt.fits
+        which are
+        filename, proposal_id, program_id, obset_id, visit_id, exptime, filters, detector, pathname
+
+    Output dict will have format (as needed by further code for creating the
+        product filenames) of:
+
+        obs_info_dict["single exposure product 00"] = {'info': '11150 A1S WFC3 IR F110W IA1S70JTQ',
+                                                        'files':['ia1s70jtq_flt.fits']}
+        obs_info_dict["single exposure product 01"] = {'info': '11150 A1S WFC3 IR F160W IA1S70JWQ',
+                                                        'files':['ia1s70jwq_flt.fits']}
+
+        obs_info_dict["filter product 00"] = {"info": '11150 A1S WFC3 IR F110W',
+                                              "files":['ia1s70jtq_flt.fits']}
+        obs_info_dict["filter product 01"] = {"info": '11150 A1S WFC3 IR F160W',
+                                              "files":['ia1s70jwq_flt.fits']}
+        obs_info_dict["total detection product 00"] = {'info': '11150 A1S WFC3 IR',
+                                                       'files':['ia1s70jtq_flt.fits',
+                                                                'ia1s70iwq_flt.fits']}
+
+    """
+    colnames = ['filename', 'proposal_id', 'program_id', 'obset_id', 'visit_id',
+                'exptime', 'filters', 'detector', 'pathname']
+    obset_table = Table.read(results, format='ascii.fast_no_header')
+    # label each column with a descriptive name
+    for col, cname in zip(obset_table.colnames, colnames):
+        obset_table[col].name = cname
+    # Add INSTRUMENT column
```

Comment:
Remove this loop if `obset_table = Table.read(results, format='ascii.fast_no_header', names=colnames)` is adopted above.

Reply:
Done.